Bottom: e345406d5bed514c0964c5576a1d4e771ab2ca7b
Top:    ed33a0c1f4bb1b9dc6e622c6a20151ed334ed2af
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-26 14:54:44 +0000

Try to prepend, not append to g_lists. This avoids wasting time traversing lists


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 494b9c7..149dcfe 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -110,19 +110,19 @@ object3d_set_appearance (object3d *object, appearance *appear)
 void
 object3d_add_edge (object3d *object, edge_ref edge)
 {
-  object->edges = g_list_append (object->edges, (void *)edge);
+  object->edges = g_list_prepend (object->edges, (void *)edge);
 }
 
 void
 object3d_add_vertex (object3d *object, vertex3d *vertex)
 {
-  object->vertices = g_list_append (object->vertices, vertex);
+  object->vertices = g_list_prepend (object->vertices, vertex);
 }
 
 void
 object3d_add_face (object3d *object, face3d *face)
 {
-  object->faces = g_list_append (object->faces, face);
+  object->faces = g_list_prepend (object->faces, face);
 }
 
 
@@ -680,7 +680,7 @@ object3d_from_contours (POLYAREA *contours,
       splice (cylinder_edges[1], SYM(cylinder_edges[1]));
     }
 
-    objects = g_list_append (objects, object);
+    objects = g_list_prepend (objects, object);
 
     link = malloc (sizeof (polygon_3d_link));
     pa->user_data = link;
@@ -1359,7 +1359,7 @@ old_object3d_from_board_outline (void)
       splice (cylinder_edges[1], SYM(cylinder_edges[1]));
     }
 
-    board_objects = g_list_append (board_objects, board_object);
+    board_objects = g_list_prepend (board_objects, board_object);
 
   } while (pa = pa->f, pa != board_outline);
